ext/date/php_date.c

```diff
@@ -2213,7 +2220,7 @@ PHPAPI int php_date_initialize(php_date_obj *dateobj, /*const*/ char *time_str,
 
 	if (ctor && err && err->error_count) {
 		/* spit out the first library error message, at least */
-		php_error_docref(NULL, E_WARNING, "Failed to parse time string (%s) at position %d (%c): %s", time_str,
+		php_error_docref(NULL, E_WARNING, "Failed to parse time string (%.*s) at position %d (%c): %s", (int) time_str_len, time_str,
```

Comment:
This is pretty unusual, and I'm not sure it's conforming to pass NULL even if the width is zero. I'd suggest `time_str ? time_str : ""` instead.

Reply:
I have done it on previous code, I just consider that the incoming may be a non-zero-termination string (of course it does not exist in the PHP kernel)
I will remove `time_str_len` here